geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -112,9 +112,11 @@ trait KafkaDataStoreSchemaManager extends DataStore with LazyLogging {
     }.getOrElse(None)
   }
 
-  override def getNames: util.List[Name] =
-    zkUtils.zkClient.getChildren(zkPath).asScala.map(name => new NameImpl(name) : Name).asJava
-
+  override def getNames: util.List[Name] = {
+    zkUtils.zkClient.getChildren(zkPath).asScala
+      .filter(name => zkUtils.zkClient.exists(getTopicPath(name)))
```

Comment:
Do we have (or should we have) any concerns about dead topic names accumulating in Zookeeper?